src/System.CommandLine/RootCommand.cs

```diff
@@ -23,7 +23,17 @@ public class RootCommand : Command
                 LazyThreadSafetyMode.PublicationOnly);
 
         private static readonly Lazy<string> _executablePath =
-            new(() => GetAssembly().Location,
+            new(() =>
+                {
+#if NET6_0_OR_GREATER
+                    var al = Environment.ProcessPath ?? "";
```

Comment:
Is it OK for this to return `C:\Program Files\dotnet\dotnet.exe` when someone uses `dotnet.exe MyApp.dll` to invoke their app?

Reply:
One option is to use Assembly.Location, and check it for `IsNullOrEmpty`. If it is empty, then assume you are in a single-file executable and fallback to ProcessPath.

For example:

https://github.com/dotnet/runtime/blob/8447c80431f091c1eb51a1468dce7638d13f7e1a/src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextLoader.cs#L134-L140

Reply:
The problem I'm seeing is that `ProcessPath` is returning just `dotnet` in some cases rather than the full path.

Reply:
When? On Windows or other?

If you are seeing that, can you log an issue in dotnet/runtime? I believe that shouldn't happen.